slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -329,6 +330,111 @@ void SLAMServiceImpl::SetUpMapBuilder() {
 }
 
 std::string SLAMServiceImpl::GetLatestJpegMapString(bool add_pose_marker) {
+    try {
+        cartographer::io::PaintSubmapSlicesResult painted_slices =
+            GetLatestPaintedMapSlices();
+    } catch (std::exception &e) {
+        // jpeg will be empty which captures the same error
+        return "";
+    }
+    cartographer::io::PaintSubmapSlicesResult painted_slices =
+        GetLatestPaintedMapSlices();
+    if (add_pose_marker) {
+        PaintMarker(painted_slices);
+    }
+
+    auto image = io::Image(std::move(painted_slices.surface));
+    return image.WriteJpegToString(50);
+}
+
+bool SLAMServiceImpl::GetLatestPointCloudMapString(std::string &pointcloud) {
+    try {
+        cartographer::io::PaintSubmapSlicesResult painted_slices =
+            GetLatestPaintedMapSlices();
+    } catch (std::exception &e) {
+        // pointcloud will be empty which captures the error
+        return false;
+    }
+    cartographer::io::PaintSubmapSlicesResult painted_slices =
+        GetLatestPaintedMapSlices();
```

Comment:
Same here. I think you meant to declare painted_slices outside of the try/catch, and then set it inside the try/catch.